core/src/transaction_weighting_stage.rs

```diff
@@ -0,0 +1,77 @@
+use {
+    crossbeam_channel::{Receiver, RecvTimeoutError, Sender},
+    rayon::prelude::*,
+    solana_gossip::cluster_info::ClusterInfo,
+    solana_perf::packet::PacketBatch,
+    solana_runtime::bank_forks::BankForks,
+    solana_streamer::streamer::{self, StreamerError},
+    std::{
+        collections::HashMap,
+        net::IpAddr,
+        sync::{Arc, RwLock},
+        thread::{self, Builder, JoinHandle},
+        time::Instant,
+    },
+};
+
+pub struct TransactionWeightStage {
+    thread_hdl: JoinHandle<()>,
+}
+
+impl TransactionWeightStage {
+    pub fn new(
+        packet_receiver: Receiver<PacketBatch>,
+        sender: Sender<Vec<PacketBatch>>,
+        bank_forks: Arc<RwLock<BankForks>>,
+        cluster_info: Arc<ClusterInfo>,
+    ) -> Self {
+        let thread_hdl = Builder::new()
+            .name("sol-tx-weight".to_string())
+            .spawn(move || {
+                let mut last_stakes = Instant::now();
+                let mut ip_to_stake: HashMap<IpAddr, u64> = HashMap::new();
+                loop {
+                    if last_stakes.elapsed().as_millis() > 1000 {
+                        let root_bank = bank_forks.read().unwrap().root_bank();
+                        let staked_nodes = root_bank.staked_nodes();
+                        ip_to_stake = cluster_info
+                            .tvu_peers()
```

Comment:
grabbing the tvu peers on every iteration here will incur 
1. unnecessary contention on the gossip lock
2.  scan through gossip for all the peers info

The more performant option is to use the `ClusterNodesCache` that @behzadnouri so helpfully added: https://github.com/solana-labs/solana/blob/8b66625c957f0e44598e1ddf647a9b5733946fca/core/src/broadcast_stage.rs#L404